common/src/Microsoft.Azure.IIoT.Core/src/Messaging/EventHub/Services/EventHubDeviceEventHandler.cs

```diff
@@ -36,17 +36,27 @@ public sealed class EventHubDeviceEventHandler : IEventHandler {
             // try to get event's properties 
             properties.TryGetValue(CommonProperties.DeviceId, out var deviceId);
             properties.TryGetValue(CommonProperties.ModuleId, out var moduleId);
-
+            var handled = false;
             foreach (var handler in _handlers.Values) {
+                if (!_used.Contains(handler)){
+                    _used.Add(handler);
+                }
                 await handler.HandleAsync(deviceId, moduleId, eventData, properties, checkpoint);
-                _used.Add(handler);
+                handled = true;
+            }
+
+            if (!handled && _unknown != null) {
+                // From a device, but does not have any event schema or message schema
+                await _unknown.HandleAsync(eventData, properties);
             }
         }
 
         /// <inheritdoc/>
         public async Task OnBatchCompleteAsync() {
             foreach (var handler in _used.ToList()) {
-                await Try.Async(handler.OnBatchCompleteAsync);
+                if (handler != null) { 
```

Comment:
We should look into why this can be null.   And then also fix that root cause.

Reply:
I'll add a bug to track this.